src/query/functions/temporal/base_test.go

```diff
@@ -351,6 +353,158 @@ func setup(numBlocks int, duration time.Duration, nextBound int) *testContext {
 	}
 }
 
+// TestBaseWithDownstreamError checks that we handle errors from blocks correctly
+func TestBaseWithDownstreamError(t *testing.T) {
+	numBlocks := 2
+	tc := setup(numBlocks, 5*time.Minute, 1)
+
+	testErr := errors.New("test err")
+	errBlock := blockWithDownstreamErr{Block: tc.Blocks[1], Err: testErr}
+
+	require.NoError(t, tc.Node.Process(models.NoopQueryContext(), parser.NodeID(0), errBlock))
+
+	err := tc.Node.Process(models.NoopQueryContext(), parser.NodeID(0), tc.Blocks[0])
+	require.EqualError(t, err, testErr.Error())
+}
+
+// Types for TestBaseWithDownstreamError
```

Comment:
nit: I think may be a bit clearer to group a bunch of these test types together? Up to you

Reply:
Personally I like them closer to where they're used, since they're effectively part of the particular tests (I'd put them inside the tests if I could, but alas, no method definitions within functions :/ )

Reply:
Yeah totally fair; I guess the tradeoff is extra clarity for particular tests when the utils are defined next to them or extra clarity for what the tests themselves cover when they're grouped

Either way super minor nitpick haha